docs/04-resources/04-tutorials/rootstock-go.md

```diff
@@ -0,0 +1,440 @@
+sidebar_position: 2
+tags: [rsk, rootstock, go, golang, go-ethereum, tutorials, resources, smart contracts]
+description: "Go is an easy to learn which combines speed and concurrency. It is widely used in web3 development, especially in the ethereum ecosystem."
+---
+
+## Prerequisites
+- Go
+    - Install the latest version of Go from [the official website](https://go.dev/).
+    To confirm it is installed, run the following command on your terminal. It should provide a valid output with your current go version. The version used in this tutorial is `1.24.1`.
+    ```bash
+    go version
+    ```
+- Valid rootstock wallet (with test RBTC from a [faucet](https://dev.rootstock.io/dev-tools/additional-tools/#faucets)) and [API key](https://dev.rootstock.io/developers/rpc-api/rootstock/setup/)
+
+
+
+## Getting Started
+Create a new folder for the project, rsk-go. Enter that folder and initialize the project from the command line.
+```bash
+go mod init rsk-go
+```
+This creates a `go.mod` file in the root directory, which holds your project's dependencies.
+
+Install the [`go-ethereum`](https://geth.ethereum.org/) package.
+<br/>Go-ethereum is a library that contains useful helpers for interacting with evm-compatible blockchains such as rootstock.
+```bash
+go get github.com/ethereum/go-ethereum
+```
+This adds the go-ethereum package to your `go.mod` file and automatically generates a `go.sum` file to the same directory, which stores extra data about your project's dependencies.
+
+```go
+// go.mod
+module rsk-go
+
+go 1.24.1
+
+require github.com/ethereum/go-ethereum v1.15.5
+```
+
+:::info[Info]
+Observe the changing imports as you paste the code.
+:::
+
+## Connect to a Rootstock node
+To confirm that we are actually connected to the rootstock network, we can write code to retrieve data from the blockchain, in this case, get the latest block.
+Create a `main.go` file in the same directory and paste this code inside it.
+
+```go
+//main.go
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+func main() {
+	rpcURL := "https://rpc.testnet.rootstock.io/YOUR-RPC-API-KEY"
+
+	// Set up the RPC connection
+	client, err := ethclient.Dial(rpcURL)
+	if err != nil {
+		log.Fatal("Failed to connect to Ethereum node:", err)
+	}
+	defer client.Close()
+
+	// Fetch the latest block number
+	blockNumber, err := client.BlockNumber(context.Background())
+	if err != nil {
+		log.Fatal("Failed to get block number:", err)
+	}
+
+	fmt.Println("Latest block number:", blockNumber)
+}
+```
+Open a terminal in your current folder and run the code. This displays the latest block in the rootstock testnet.
+<br/>Run it again after a few seconds to confirm that the block number increases as expected.
+```bash
+go run main.go
+```
+
+## Get my RBTC balance
+Get your wallet ready. You can confirm the balance from other sources such as your metamask wallet to ensure accuracy.
```

Comment:
```suggestion
To confirm the balance from other sources such as your Metamask wallet to ensure accuracy. Replace `0xMY-WALLET-ADDRESS` with your wallet address.
```

Reply:
Add a brief explanation about why getting the wallet balance (RBTC) is important. For e.g, to write to the blockchain